### PR TITLE
Logout admin's Intercom session when admin logs out

### PR DESCRIPTION
**Pull request by rap1ds**

### CODE REVIEW

File: app/view_utils/intercom_helper.rb [hunk 1]
```diff
@@ -1,6 +1,56 @@
...
+    # findDomain: function(e) {
+    #   var t = e.match(o);
+    #   if (!t && (t = e.match(n))) {
+    #     var r = t[0];
+    #     return r = r.split(":")[0], "." + r
+    #   }
+    # }
+    # ```
+    #
+    # The regexp `o` is omited. (It seems to be IP matcher)
```
Comment (sktoiva) [inline app/view_utils/intercom_helper.rb]: omited -> omitted

File: app/view_utils/intercom_helper.rb [hunk 2]
```diff
@@ -1,6 +1,56 @@
...
+    #
+    def self.find_domain(host_with_port)
+      domain_regexp = /[^.]*\.([^.]*|..\...|...\...)$/
+
+      match = domain_regexp.match(host_with_port)
+
+      if match
```
Comment (sktoiva) [inline app/view_utils/intercom_helper.rb]: Can be removed.

File: app/view_utils/intercom_helper.rb [hunk 3]
```diff
@@ -1,6 +1,56 @@
...
+        ".#{match[0].split(":")[0]}"
+      end
+    end
+
+    def self.intercom_shutdown(session, cookies, host_with_port)
+      domain = find_domain(host_with_port)
+
+      if domain
```
Comment (sktoiva) [inline app/view_utils/intercom_helper.rb]: Can be removed.

File: app/view_utils/intercom_helper.rb [hunk 4]
```diff
@@ -1,6 +1,51 @@
...
 
+  # Copied from
+  #
+  # https://github.com/intercom/intercom-rails/blob/1a077474f33c2629de1d1a46ff67f97c0b9a0264/lib/intercom-rails/shutdown_helper.rb
+  #
+  # ...with modifications.
+  #
+  module ShutdownHelper
+
+    LOGOUT_KEY = :prepare_intercom_shutdown
```
Comment (rap1ds) [inline app/view_utils/intercom_helper.rb]: This is not in use. Remove.

**Pull request merged**